cuebot/src/test/java/com/imageworks/spcue/test/dispatcher/HostReportHandlerTests.java

```diff
@@ -0,0 +1,125 @@
+
+/*
+ * Copyright (c) 2018 Sony Pictures Imageworks Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+
+package com.imageworks.spcue.test.dispatcher;
+
+import javax.annotation.Resource;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.imageworks.spcue.DispatchHost;
+import com.imageworks.spcue.dispatcher.Dispatcher;
+import com.imageworks.spcue.dispatcher.HostReportHandler;
+import com.imageworks.spcue.grpc.host.HardwareState;
+import com.imageworks.spcue.grpc.host.LockState;
+import com.imageworks.spcue.grpc.report.CoreDetail;
+import com.imageworks.spcue.grpc.report.HostReport;
+import com.imageworks.spcue.grpc.report.RenderHost;
+import com.imageworks.spcue.service.AdminManager;
+import com.imageworks.spcue.service.HostManager;
+import com.imageworks.spcue.test.TransactionalTest;
+import com.imageworks.spcue.util.CueUtil;
+
+import static org.junit.Assert.assertEquals;
+
+@ContextConfiguration
+public class HostReportHandlerTests extends TransactionalTest {
+
+    @Resource
+    AdminManager adminManager;
+
+    @Resource
+    HostManager hostManager;
+
+    @Resource
+    HostReportHandler hostReportHandler;
+
+    @Resource
+    Dispatcher dispatcher;
+
+    private static final String HOSTNAME = "beta";
+
+    @Before
+    public void setTestMode() {
+        dispatcher.setTestMode(true);
+    }
+
+    @Before
+    public void createHost() {
+        hostManager.createHost(getRenderHost(),
+                adminManager.findAllocationDetail("spi","general"));
+    }
+
+    public static CoreDetail getCoreDetail(int total, int idle, int booked, int locked) {
+        return CoreDetail.newBuilder()
+                .setTotalCores(total)
+                .setIdleCores(idle)
+                .setBookedCores(booked)
+                .setLockedCores(locked)
+                .build();
+    }
+
+    public DispatchHost getHost() {
+        return hostManager.findDispatchHost(HOSTNAME);
+    }
+
+    public RenderHost getRenderHost() {
```

Comment:
Do these helper methods need to be public?

Reply:
They do not. Done.